Dockerfile

```diff
@@ -21,6 +21,18 @@ ENV PATH=${PATH}:/go/src/github.com/docker/app/bin/
 ARG DEP_VERSION=v0.5.0
 RUN curl -o /usr/bin/dep -L https://github.com/golang/dep/releases/download/${DEP_VERSION}/dep-linux-amd64 && \
     chmod +x /usr/bin/dep
+ARG GOTESTSUM_VERSION=v0.3.4
+RUN mkdir $GOPATH/src/gotest.tools && \
+    git clone -q https://github.com/gotestyourself/gotestsum $GOPATH/src/gotest.tools/gotestsum && \
+    cd $GOPATH/src/gotest.tools/gotestsum && \
+    git -C $GOPATH/src/gotest.tools/gotestsum checkout -q $GOTESTSUM_VERSION && \
+    GO111MODULE=on GOOS=linux   go build -o /usr/local/bin/gotestsum-linux       gotest.tools/gotestsum && \
+    GO111MODULE=on GOOS=darwin  go build -o /usr/local/bin/gotestsum-darwin      gotest.tools/gotestsum && \
+    GO111MODULE=on GOOS=windows go build -o /usr/local/bin/gotestsum-windows.exe gotest.tools/gotestsum && \
+    ln -s gotestsum-linux /usr/local/bin/gotestsum
+RUN GOOS=linux   go build -o /usr/local/bin/test2json-linux       cmd/test2json && \
```

Comment:
I'm really missing something here, I don't understand where this `cmd/test2json` come from 🤔

Reply:
```
commit 207be839fbf729c74c3d39bdadffbd5882fc2f30
Author: Ian Campbell <ijc@docker.com>
Date:   Mon Apr 1 17:09:01 2019 +0100

    ci: run e2e tests via gotestsum and record junit result
    
    As well as gotestsum we now also need to build `test2json` because test
    binaries build with `go test -c` do not support the `-test` option. That option
    just wraps the test run in the `test2json` code which translates from the
    standard output syntax to json. (See https://github.com/golang/go/issues/31051
    for more info). This uses `cmd/test2json` the source of which is part of the go
    distribution and is therefore available in the base images.
```

Reply:
Ooooh THAT was a tricky one ! Ok then 👍

Reply:
Maybe add a comment then?
`# cmd/test2json is part of golang distribution`

Reply:
Good idea. I just `--amend`ed with:
```diff
$ git diff
diff --git a/Dockerfile b/Dockerfile
index fb3b5e3c..c1700ec2 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -30,6 +30,8 @@ RUN mkdir $GOPATH/src/gotest.tools && \
     GO111MODULE=on GOOS=darwin  go build -o /usr/local/bin/gotestsum-darwin      gotest.tools/gotestsum && \
     GO111MODULE=on GOOS=windows go build -o /usr/local/bin/gotestsum-windows.exe gotest.tools/gotestsum && \
     ln -s gotestsum-linux /usr/local/bin/gotestsum
+# Source for cmd/test2json is part of the Go distribution and is
+# therefore available in the base image.
 RUN GOOS=linux   go build -o /usr/local/bin/test2json-linux       cmd/test2json && \
     GOOS=darwin  go build -o /usr/local/bin/test2json-darwin      cmd/test2json && \
     GOOS=windows go build -o /usr/local/bin/test2json-windows.exe cmd/test2json
```